src/client/pages/Offer/index.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  useHistory,
+  useRouteMatch,
+  RouteComponentProps,
+  Redirect,
+} from 'react-router'
+import { LoadingPage } from 'components/LoadingPage'
+import { TopBar } from 'components/TopBar'
+import { Page } from 'components/utils/Page'
+import { useQuoteCartQuery, QuoteBundleVariant } from 'data/graphql'
+import { useVariation, Variation } from 'utils/hooks/useVariation'
+import { trackOfferGTM, EventName } from 'utils/tracking/gtm'
+import { getUtmParamsFromCookie, TrackAction } from 'utils/tracking/tracking'
+import { localePathPattern } from 'l10n/localePathPattern'
+import { Features, useFeature } from 'utils/hooks/useFeature'
+import {
+  useContractTypes,
+  useContractTypesFromQueryParam,
+} from 'utils/hooks/useContractTypes'
+import { useCurrentLocale } from 'l10n/useCurrentLocale'
+import { LanguagePicker } from '../Embark/LanguagePicker'
+import {
+  getOfferData,
+  getUniqueQuotesFromVariantList,
+  getBundleVariantFromContractTypes,
+} from '../OfferNew/utils'
+import { AppPromotionSection } from '../OfferNew/AppPromotionSection'
+import { Checkout } from '../OfferNew/Checkout'
+import { FaqSection } from '../OfferNew/FaqSection'
+import { Introduction } from '../OfferNew/Introduction'
+import { Perils } from '../OfferNew/Perils'
+import { InsuranceSelector } from '../OfferNew/InsuranceSelector'
+
+const createToggleCheckout = (history: History<any>, locale?: string) => (
+  isOpen: boolean,
+) => {
+  if (isOpen) {
+    history.push(`/${locale}/new-member/sign`)
+  } else {
+    history.goBack()
+  }
+}
+
+const getContractTypesFromBundleVariant = (bundleVariant: QuoteBundleVariant) =>
+  bundleVariant.bundle.quotes.map((quote) => quote.typeOfContract)
+
+type OfferPageProps = RouteComponentProps<{ id: string }>
+
+export const OfferPage = ({
+  match: {
+    params: { id: quoteCartId },
+  },
+}: OfferPageProps) => {
+  const { isoLocale, path: pathLocale } = useCurrentLocale()
+  const variation = useVariation()
+  const [isInsuranceToggleEnabled, isQuoteCartEnabled] = useFeature([
+    Features.OFFER_PAGE_INSURANCE_TOGGLE,
+    Features.QUOTE_CART_API,
+  ])
+
+  const initialContractTypes = useContractTypesFromQueryParam()
+  const [selectedContractTypes, setSelectedContractTypes] = useContractTypes(
+    initialContractTypes ?? undefined,
+  )
+
+  const history = useHistory()
+
+  const {
+    data: quoteCartData,
+    loading: isLoadingQuoteCart,
+    refetch: refetchQuoteCart,
+    error: quoteCartError,
+  } = useQuoteCartQuery({
+    variables: {
+      id: quoteCartId,
+      locale: isoLocale,
+    },
+  })
+
+  const bundleData = quoteCartData?.quoteCart.bundle
+  const redeemedCampaigns = [quoteCartData?.quoteCart.campaign]
+
+  if (quoteCartError) {
+    throw new Error(`Error fetching quote cart: ${quoteCartId}.`)
+  }
+
+  if (!isLoadingQuoteCart && !bundleData) {
+    throw new Error(
+      `No bundle returned to show offer with (quote cart ID=${quoteCartId}).`,
+    )
+  }
```

Comment:
We need error handling for these cases but it's not worse than what we have today.